modules/navier_stokes/src/postprocessors/VolumetricFlowRate.C

```diff
@@ -16,23 +17,122 @@ InputParameters
 VolumetricFlowRate::validParams()
 {
   InputParameters params = SideIntegralPostprocessor::validParams();
-  params.addClassDescription("Computes the volumetric flow rate through a boundary.");
-  params.addRequiredCoupledVar("vel_x", "The x velocity");
-  params.addCoupledVar("vel_y", 0, "The y velocity");
-  params.addCoupledVar("vel_z", 0, "The z velocity");
+  params.addClassDescription(
+      "Computes the volumetric flow rate of an advected quantity through an external boundary.");
+  params.addParam<bool>("fv", false, "Whether finite volume variables are used");
+  params.addRequiredCoupledVar("vel_x", "The x-axis velocity");
+  params.addCoupledVar("vel_y", 0, "The y-axis velocity");
+  params.addCoupledVar("vel_z", 0, "The z-axis velocity");
+  params.addCoupledVar(
+      "advected_variable", 0, "The advected variable quantity of which to study the flow");
+  params.addParam<MaterialPropertyName>(
+      "advected_mat_prop", 0, "The advected material property of which to study the flow");
+  MooseEnum advected_interp_method("average upwind", "upwind");
+  params.addParam<MooseEnum>("advected_interp_method",
+                             advected_interp_method,
+                             "The interpolation to use for the advected quantity. Options are "
+                             "'upwind' and 'average', with the default being 'upwind'.");
   return params;
 }
 
 VolumetricFlowRate::VolumetricFlowRate(const InputParameters & parameters)
   : SideIntegralPostprocessor(parameters),
+    _fv(getParam<bool>("fv")),
     _vel_x(coupledValue("vel_x")),
     _vel_y(coupledValue("vel_y")),
-    _vel_z(coupledValue("vel_z"))
+    _vel_z(coupledValue("vel_z")),
+    _fv_vel_x(dynamic_cast<const MooseVariableFV<Real> *>(getFieldVar("vel_x", 0))),
+    _fv_vel_y(dynamic_cast<const MooseVariableFV<Real> *>(getFieldVar("vel_y", 0))),
+    _fv_vel_z(dynamic_cast<const MooseVariableFV<Real> *>(getFieldVar("vel_z", 0))),
+    _advected_variable_supplied(parameters.isParamSetByUser("advected_variable")),
```

Comment:
I'm not going to make you change it, but in the future I believe we have a `MooseObject::isParamSetByUser` so you don't have to call through the `parameters` object

Reply:
ok I ll catch it in PINSFV then